### PR TITLE
#751 Documentation of functions

### PR DESCRIPTION
**Pull request by CREITZ25**
#751 Documentation of functions

### CODE REVIEW

**Comment by CREITZ25** [comment]
The following functions are not documented:

1. Maven specific functions
doArrayContainsItemWithPrefix
doMavenArchetype
doMavenGetLatestVersion
doUpgradeMavenArtifact

2. Workspace Management
doConfigureWorkspace
doCreateIdeScript
doDevonCommandAllWorkspaces

3. doDownload with the version functionality
doDownload
doGetNextVersion
doVersionCompare

**Comment by CREITZ25** [comment]
Adding documentation of functions doGetNextVersion and doVersionCompare

**Comment by CREITZ25** [comment]
The following functions are not documented:

Maven specific functions
doArrayContainsItemWithPrefix
doMavenArchetype
doMavenGetLatestVersion
doUpgradeMavenArtifact

Workspace Management
doConfigureWorkspace
doCreateIdeScript

doDownload with the version functionality
doDownload

**Comment by CREITZ25** [comment]
Correcting typing errors and adding the three funktions doError, doSuccess and doQuestion

**Comment by cinnamon-coder-hub** [comment]
> The following functions are not documented:
> 
> Maven specific functions doArrayContainsItemWithPrefix doMavenArchetype doMavenGetLatestVersion doUpgradeMavenArtifact
> 
> Workspace Management doConfigureWorkspace doCreateIdeScript
> 
> doDownload with the version functionality doDownload

@CREITZ25, there was some additional feedback that I wanted to mention stating, that it would be useful to include the functions not mentioned inside the documentation yet with a "to do" comment, like 
= doConfigureWorkspace
to do

as an example. 
Also it was mentioned that it would be good to group all functions and some minor structural changes, 
like just using "== doSuccess" instead of "== doSuccess ==". 
We can also speak through the feedback this week : )

**Review by cinnamon-coder-hub — APPROVED** [review]

**Comment by hohwille** [comment]
Can we get the hierarchy fixed (heading levels aligned 1. title, 1.1. cateogy 1.1.1, function 1.1.2 function, 1.2 category, ... rather than 1.1 category, 1.1.1. function, ... 1.x. function, ...) and take this out of draft so I can merge this?
I am happy to merge with all the TODOs and we can have a category TODO with all functions not yet categorized so I can later address the TODOs. But only if it is merged, new PRs adding new functions can also add their documentation.

File: documentation/functions.asciidoc [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+= doReplaceExtractedFile
+The doReplaceExtractedFile function is called with the three parameters path of the file or directory to be moved, the target directory and a backup directory. The function creates missing directories, makes a backup from the target directory to the backup directory. Then the files specified in the first parameter are moved to the target directory.
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply documentation/functions.asciidoc:180]: done

File: documentation/functions.asciidoc [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+The function doRequireWsl checks whether WSL for Windows is installed and whether it is activated and issues a corresponding message via doFail if one of the two requirements is not met.
+
+= doRunCommand
+The doRunCommand function can be given 3 parameters, the command to be executed, a message to be printed and the directory in which the command is to be executed. First, the function is checked whether the specified directory exists. If the directory does not exist, it will take the current directory. Then the command is executed and depending on the return value of the command, a message is output that starts with "Succeed" or "Failed".
+
+= doRunConfigurator
+The doRunConfigurator function transfers all jar files in the DEVON_IDE_HOME/scripts/lib directory separated by ":" as a class path to the Java command and executes the Java program specified as a parameter.
+
+= doUnzip
+The doUnzip function checks if an unzip program is installed, and if not, installs the program. The file specified in the first parameter is then unpacked into the directory specified in the second parameter.
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply documentation/functions.asciidoc:198]: Function is corrected

**Pull request merged**